src/pydocstyle/checker.py

```diff
@@ -204,17 +202,23 @@ def check_docstring_missing(self, definition, docstring):
                 Module: violations.D100,
                 Class: violations.D101,
                 NestedClass: violations.D106,
-                Method: (
-                    lambda: violations.D105()
-                    if definition.is_magic
+                Method: lambda: violations.D105()
+                if definition.is_magic
+                else (
+                    violations.D107()
+                    if definition.is_init
                     else (
-                        violations.D107()
-                        if definition.is_init
-                        else violations.D102()
+                        violations.D102()
+                        if not definition.is_overload
+                        else None
                     )
                 ),
-                Function: violations.D103,
                 NestedFunction: violations.D103,
```

Comment:
I'm surprised no changes were needed here?

Reply:
Yes, me too but it looks like a "nested function" is a function that isn't public.
as its composed from the original function class then no changes are needed.
Well, I think this is the case, not an expert on python, but it works :)